packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1227,6 +1270,112 @@ class RenderEditable extends RenderBox {
     }
   }
 
+  /// Sets the screen position of the floating cursor and the text position
+  /// closest to the cursor.
+  void setFloatingCursor(FloatingCursorDragState state, Offset boundedOffset, TextPosition lastTextPosition, { double resetLerpValue }) {
+    assert(state != null);
+    assert(boundedOffset != null);
+    assert(lastTextPosition != null);
+    if (state == FloatingCursorDragState.Start) {
+      _relativeOrigin = const Offset(0, 0);
+      _previousOffset = null;
+      _resetOriginOnBottom = false;
+      _resetOriginOnTop = false;
+      _resetOriginOnRight = false;
+      _resetOriginOnBottom = false;
+    }
+    _floatingCursorOn = state != FloatingCursorDragState.End;
+    _resetLerpValue = resetLerpValue;
+    if(_floatingCursorOn) {
+      _floatingCursorOffset = boundedOffset;
+      _floatingCursorTextPosition = lastTextPosition;
+    }
+    markNeedsPaint();
+  }
+
+  void _paintFloatingCaret(Canvas canvas, Offset effectiveOffset) {
+    assert(_textLayoutLastWidth == constraints.maxWidth);
+    final Paint paint = Paint()
+      ..color = _cursorColor;
+
+    double sizeAdjustmentX = _kFloatingCaretSizeOffset.dx;
+    double sizeAdjustmentY = _kFloatingCaretSizeOffset.dy;
+
+    if(_resetLerpValue != null) {
+      sizeAdjustmentX = ui.lerpDouble(sizeAdjustmentX, 0, _resetLerpValue);
+      sizeAdjustmentY = ui.lerpDouble(sizeAdjustmentY, 0, _resetLerpValue);
+    }
+
+    final Rect caretPrototype = Rect.fromLTRB(_caretPrototype.left - sizeAdjustmentX,
```

Comment:
floatingCaretProtoType? So people looking for things via string find won't have to manually sift through things that are 2 different things.